### PR TITLE
Add whitelist information to dendrite buildkite annotation

### PR DESCRIPTION
**Pull request by anoadragon453**
Takes the output of dendrite's `show-expected-fail-tests.sh` script and appends it to the buildkite annotation output.

This allows us to see which tests we need to add/remove to/from the whitelist/blacklist without having to dig down in the logs.

### CODE REVIEW

File: scripts/dendrite_sytest.sh [hunk 1]
```diff
@@ -62,6 +62,9 @@ rsync -r --ignore-missing-args --min-size=1B -av /work/server-0 /work/server-1 /
 if [ $TEST_STATUS -ne 0 ]; then
     # Build the annotation
     perl /sytest/scripts/format_tap.pl /logs/results.tap "$BUILDKITE_LABEL" >/logs/annotate.md
+    # If show-expected-fail-tests logged something, put it in the annotation
+    # We want nice output though, so surround the text output in code tags
```
Comment (richvdh) [inline scripts/dendrite_sytest.sh]: > surround the text output in code tags

where does this happen?

Comment (anoadragon453) [reply scripts/dendrite_sytest.sh:66]: Wow, this totally fell off my radar.

It's supposed to happen inside of `format_tap.pl`, which was *probably* supposed to be part of this PR, but perhaps I didn't push that change?

Time to go inspect my desktop...

Comment (anoadragon453) [reply scripts/dendrite_sytest.sh:66]: Aha, no they're here! https://github.com/matrix-org/dendrite/pull/870/files

Comment (richvdh) [reply scripts/dendrite_sytest.sh:66]: ok, but still, that means this comment is in the wrong place?

File: scripts/dendrite_sytest.sh [hunk 2]
```diff
@@ -52,7 +52,7 @@ fi
 
 # Check for new tests to be added to the test whitelist
 /src/show-expected-fail-tests.sh /logs/results.tap /src/sytest-whitelist \
-    /src/sytest-blacklist || TEST_STATUS=$?
+    /src/sytest-blacklist > /src/test-whitelist.log || TEST_STATUS=$?
```
Comment (richvdh) [inline scripts/dendrite_sytest.sh:55]: could this file have a name which better represents what it does?

Also: I'd recommend against writing to `/src`. It means that we'll be writing stuff to somebody's working copy of dendrite, and means we can't run it against a read-only mount of the dendrite source.

File: scripts/dendrite_sytest.sh (hunk 1 above)
Comment (anoadragon453) [reply scripts/dendrite_sytest.sh:66]: Ah, yes quite.

**Pull request merged**